lib/Cake/Controller/Component/Auth/BasicAuthenticate.php

```diff
@@ -83,8 +83,8 @@ public function authenticate(CakeRequest $request, CakeResponse $response) {
  */
 	public function getUser(CakeRequest $request) {
 		if(!isset($_SERVER['PHP_AUTH_USER'])) {
-		    if (isset($_SERVER['HTTP_AUTHORIZATION']) && strlen($_SERVER['HTTP_AUTHORIZATION']) > 0 && strpos($_SERVER['HTTP_AUTHORIZATION'], 'basic') !== false) {
-		        list($_SERVER['PHP_AUTH_USER'], $_SERVER['PHP_AUTH_PW']) = explode(':', base64_decode(substr($_SERVER['HTTP_AUTHORIZATION'], 6)));
+			if ($request->header('Authorization') !== false && strlen($request->header('Authorization')) > 0 && strpos($request->header('Authorization'), 'basic') !== false) {
```

Comment:
I would assign ``$request->header('Authorization')`` to a variable and reuse it, instead of calling it over and over.